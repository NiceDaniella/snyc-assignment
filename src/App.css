.App {
  text-align: center;
}
form {
  display: flex;
  justify-content: center;
  margin: 20px;
  align-items: center;
  background: #5bb8c3;
  border: 1px solid #e6e6e6;
  border-radius: 4px;
  padding: 10px;
  box-shadow: rgb(158, 158, 158) 6px 10px 31px -12px;
}
label {
  color: #434343;
}
input {
  width: 150px;
  height: 24px;
  margin-right: 20px;
  border: 1px solid #cccccc;
  border-radius: 4px;
}
div {
  margin: 10px;
}
button {
  width: 180px;
  height: 32px;
  color: #ffffff;
  border-radius: 4px;
  background-color: #5e5e5ed4;
  border: 1px solid #a6a6a6;
  font-size: 14px;
  cursor: pointer;
}
p {
  color: #434343;
  font-weight: bold;
  font-size: 20px;
}
