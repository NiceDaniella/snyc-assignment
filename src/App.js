import React, { useState } from "react";
import "./App.css";
import axios from "axios";
import Tree from "react-d3-tree";

function App() {
  const [data, setData] = useState();
  const [packageName, setPackageName] = useState("");
  const [version, setVersion] = useState("");

  function handleInputName(e) {
    setPackageName(e.target.value);
  }

  function handleInputVersion(e) {
    setVersion(e.target.value);
  }
  async function getDependenciesApiCall() {
    const response = await axios.post("/get_dependencies", {
      packageName,
      version
    });
    if (response.status !== 200) {
      throw Error("Client was unable to fetch data" + response.statusText);
    }
    return response.data;
  }
  function handleSubmit(e) {
    e.preventDefault();
    getDependenciesApiCall(packageName, version)
      .then(res => {
        setData(res);
        setPackageName("");
        setVersion("");
      })
      .catch(err => console.log("API call to fetch data failed", err));
  }

  return (
    <div className="App">
      <form onSubmit={handleSubmit} method="post">
        <div>
          <label data-testid="package-name-label">Enter package name: </label>
          <input
            data-testid="package-name-input"
            type="text"
            name="packageName"
            id="packageName"
            value={packageName}
            onChange={e => handleInputName(e)}
            required
          />
          <label data-testid="version-label">Enter version: </label>
          <input
            data-testid="version-input"
            type="text"
            name="version"
            id="version"
            value={version}
            onChange={e => handleInputVersion(e)}
            required
          />
        </div>
        <button data-testid="submit-button" type="submit">
          Get dependencies
        </button>
      </form>
      <p data-testid="display-graph">Dependencies graph:</p>
      {data && (
        <div id="treeWrapper" style={{ width: "100%", height: "600px" }}>
          <Tree data={data} />
        </div>
      )}
    </div>
  );
}

export default App;
