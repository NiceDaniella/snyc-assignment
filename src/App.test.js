import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import axiosMock from "axios";
import App from "./App.js";

jest.mock("axios");

afterEach(cleanup);

it("fetches and displays data", async () => {
  const { getByText } = render(<App />);

  axiosMock.post.mockResolvedValueOnce({
    data: {
      name: "react",
      children: []
    }
  });

  fireEvent.click(getByText("Get dependencies"));

  expect(axiosMock.post).toHaveBeenCalledTimes(1);
  expect(axiosMock.post).toHaveBeenCalledWith("/get_dependencies", {
    packageName: "",
    version: ""
  });
});

it("Input values updates on change", () => {
  const { getByTestId } = render(<App />);
  const packageNameInput = getByTestId("package-name-input");
  const versionInput = getByTestId("version-input");

  fireEvent.change(packageNameInput, { target: { value: "test" } });
  fireEvent.change(versionInput, { target: { value: "version" } });

  expect(packageNameInput.value).toBe("test");
  expect(versionInput.value).toBe("version");
});

it("renders form elements", () => {
  const { queryByTestId } = render(<App />);

  expect(queryByTestId("package-name-label")).toBeTruthy();
  expect(queryByTestId("package-name-input")).toBeTruthy();
  expect(queryByTestId("version-label")).toBeTruthy();
  expect(queryByTestId("version-input")).toBeTruthy();
  expect(queryByTestId("submit-button")).toBeTruthy();
  expect(queryByTestId("display-graph")).toBeTruthy();
});

it("App component snapshot", () => {
  const container = render(<App />);
  expect(container).toMatchSnapshot();
});
