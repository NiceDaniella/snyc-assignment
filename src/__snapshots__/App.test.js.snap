// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <form
          method="post"
        >
          <div>
            <label
              data-testid="package-name-label"
            >
              Enter package name: 
            </label>
            <input
              data-testid="package-name-input"
              id="packageName"
              name="packageName"
              required=""
              type="text"
              value=""
            />
            <label
              data-testid="version-label"
            >
              Enter version: 
            </label>
            <input
              data-testid="version-input"
              id="version"
              name="version"
              required=""
              type="text"
              value=""
            />
          </div>
          <button
            data-testid="submit-button"
            type="submit"
          >
            Get dependencies
          </button>
        </form>
        <p
          data-testid="display-graph"
        >
          Dependencies graph:
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <form
        method="post"
      >
        <div>
          <label
            data-testid="package-name-label"
          >
            Enter package name: 
          </label>
          <input
            data-testid="package-name-input"
            id="packageName"
            name="packageName"
            required=""
            type="text"
            value=""
          />
          <label
            data-testid="version-label"
          >
            Enter version: 
          </label>
          <input
            data-testid="version-input"
            id="version"
            name="version"
            required=""
            type="text"
            value=""
          />
        </div>
        <button
          data-testid="submit-button"
          type="submit"
        >
          Get dependencies
        </button>
      </form>
      <p
        data-testid="display-graph"
      >
        Dependencies graph:
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
